### PR TITLE
Make comm even more configurable + bugfix in python rtop

### PR DESCRIPTION
**Pull request by janfeitsma**
Integration fixes in Falcons codebase.

This is ready to merge.

### CODE REVIEW

File: comm/MulticastSocket.cpp [hunk 1]
```diff
@@ -214,7 +214,17 @@ bool MulticastSocket::autoSelectInterface()
...
     ignore.insert("lo");
-    ignore.insert("enp0s31f6"); // Falcons robots: the port on top-side of CPU-box, located most inward, is reserved for multiCam
+    bool iscoach = false;
+    char buf[1024] = {0};
+    gethostname(buf, 1024);
+    iscoach = std::string(buf) == "coach";
+    // ugly HACK to survive within Falcons code context
+    // TODO: make this configurable via XML
+    // except for coach laptop, which actually must connect via enp0s31f6
+    if (!iscoach)
```
Comment (janfeitsma) [inline comm/MulticastSocket.cpp]: The objective of this PR is to get rid of this hack, bring the prioritize- and ignore lists via XML in here.

File: comm/MulticastSocket.cpp [hunk 2]
```diff
@@ -211,31 +211,25 @@ bool MulticastSocket::autoSelectInterface()
...
-    
-    // priority list: assume ad-hoc cable connection takes precedence over wifi
-    std::set<std::string> prioritize;
-    prioritize.insert("enp3s0"); // Falcons robots
-    prioritize.insert("eth0"); // Falcons devlaptlops
-    
-    // TODO: make these lists configurable via XML?
-    
     // first select from priority list, then select whichever remains
     for (int priorityLoop = 0; priorityLoop <= 1; ++priorityLoop)
```
Comment (RobBurgers) [inline comm/MulticastSocket.cpp:215]: This scores well on cyclomatic complexity...
Should this method have a unit test?

File: README.md [hunk 3]
```diff
@@ -108,7 +108,9 @@ Network elements:
 
 * MulticastAddress (M) assigned group address according to MSL rules
 * MulticastPort    (M) port to use
-* Interface        (O) interface overrule, 'auto' (default) will resolve to available wireless adapter
+* Interface        (O) interface overrule, 'auto' (default) will resolve to available adapter
+* InterfacePriorityList  (O) in case 'auto' is used above, then given sequence is used to prioritize
```
Comment (RobBurgers) [inline README.md]: IMO PriorityList and BlackList should be a subelement of Interface.

File: config/rtdb_configuration.xsd [hunk 4]
```diff
@@ -105,7 +105,21 @@
             <xs:sequence>
                   <xs:element name="MulticastAddress" type="xs:string"></xs:element>
                   <xs:element name="MulticastPort" type="xs:int"></xs:element>
-                  <xs:element name="Interface" type="xs:string" minOccurs="0"></xs:element>
+                  <xs:element name="Interface" type="xs:string" minOccurs="0" maxOccurs="1"></xs:element>
+                  <xs:element name="InterfacePriorityList" minOccurs="0" maxOccurs="1">
+                        <xs:complexType>
+                              <xs:sequence>
```
Comment (RobBurgers) [inline config/rtdb_configuration.xsd]: Use type definition for list. Same type can be used for PriorityList and BlackList.

**Pull request merged**